python/tests/unit/arcticdb/version_store/test_query_builder.py

```diff
@@ -111,6 +111,52 @@ def test_reuse_querybuilder(lmdb_version_store_tiny_segment):
     assert_frame_equal(expected, received)
 
 
+def test_reuse_querybuilder_date_range(lmdb_version_store_tiny_segment):
+    lib = lmdb_version_store_tiny_segment
+    symbol = "test_reuse_querybuilder_date_range"
+    df = pd.DataFrame(
+        {"col1": np.arange(1, 11, dtype=np.int64)}, index=pd.date_range("2000-01-01", periods=10)
+    )
+    lib.write(symbol, df)
+
+    q = QueryBuilder()
+    q = q[q["col1"].isin(2, 3, 7)]
+
+    expected_0 = df.query("col1 in [2, 3]")
+    received_0 = lib.read(symbol, date_range=(None, pd.Timestamp("2000-01-06")), query_builder=q).data
+    assert_frame_equal(expected_0, received_0)
+
+    received_1 = lib.read(symbol, date_range=(None, pd.Timestamp("2000-01-06")), query_builder=q).data
+    assert_frame_equal(expected_0, received_1)
+
```

Comment:
Would another test with a different date_range and the same query be worthwhile?